pkg/skaffold/runner/runner.go

```diff
@@ -167,74 +166,52 @@ func (r *SkaffoldRunner) Run(ctx context.Context, out io.Writer, artifacts []*v1
 // Dev watches for changes and runs the skaffold build and deploy
 // pipeline until interrrupted by the user.
 func (r *SkaffoldRunner) Dev(ctx context.Context, out io.Writer, artifacts []*v1alpha2.Artifact) ([]build.Artifact, error) {
-	depMap, err := r.DependencyMapFactory(artifacts)
-	if err != nil {
-		return nil, errors.Wrap(err, "getting path to dependency map")
-	}
-
-	watcher, err := r.WatcherFactory(depMap.Paths())
-	if err != nil {
-		return nil, errors.Wrap(err, "creating watcher")
-	}
+	imageList := kubernetes.NewImageList()
+	colorPicker := kubernetes.NewColorPicker(artifacts)
+	logger := kubernetes.NewLogAggregator(out, imageList, colorPicker)
 
 	deployDeps, err := r.Dependencies()
```

Comment:
I think we'll have to introduce something similar for deploy dependencies so that they are recalculated during the loop.

Reply:
Right now these are mostly static or unimplemented - except kubectl accepts a glob pattern, which can change if new untracked files are added that satisfy the pattern. r.Dependencies() can act like the DependencyMap did for artifacts.

Reply:
Yes, I’d rather handle that in another PR because it’s a bit more complicated. New manifests could be in a folder that’s not watched.